### PR TITLE
Spike/sti

### PR DESCRIPTION
**Pull request by glassresistor**
This is ready to go.  The migrations test is failing but the code works.

The messages/_message template had to be readded for use by the message broadcast job but I think we can fix that later.

### CODE REVIEW

File: app/models/message.rb [hunk 1]
```diff
@@ -183,15 +183,15 @@ def analytics_tracker_data
   end
 
   def marker?
-    marker_type.present? && marker_type != AUTO_COURT_REMINDER
+    ![AUTO_COURT_REMINDER, TEXT_MESSAGE].include? type
```
Comment (tmaybe) [inline app/models/message.rb]: can do `[AUTO_COURT_REMINDER, TEXT_MESSAGE].exclude? type`

File: app/views/court_reminders/_court_reminders.erb [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    </div>
+  </div>
+  <p class="message--label <%= court_reminder.twilio_status %>">
+  <% if court_reminder.twilio_status == 'undelivered' %>
+    <%= t('message.status.undelivered') %>
+  <% else %>
+    <%= court_reminder.twilio_status %> <%= court_reminder.send_at.to_s(:message_timestamp) %>
+  <% end %>
+  </p>
+</div>
```
Comment (tmaybe) [inline app/views/court_reminders/_court_reminders.erb]: Should be consistent on pluralization in the naming of these files; change the name of this file to `_court_reminder.erb`.

File: spec/factories/text_message.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+    reporting_relationship { create :reporting_relationship }
+    body { "i am a message #{SecureRandom.hex(17)}" }
+    sequence(:number_from) { Faker::PhoneNumber.cell_phone }
+    sequence(:number_to) { Faker::PhoneNumber.cell_phone }
+    inbound { [true, false].sample }
+    sequence(:twilio_sid) { |n| (n.to_s + SecureRandom.hex(17))[0..33] }
+    twilio_status { %w[accepted queued sending sent receiving received delivered undelivered failed].sample }
+    send_at { Time.current }
+  end
+end
```
Comment (tmaybe) [inline spec/factories/text_message.rb:12]: None of the new factories except this one should need `number_from`, `number_to`, `twilio_sid`, `twilio_status`

File: spec/models/department_spec.rb [hunk 4]
```diff
@@ -67,12 +67,12 @@
...
+    let!(:transfers1) { create_list :text_message, 2, send_at: now - 1.day, type: Message::MARKER_TRANSFER, reporting_relationship: user1.reporting_relationships.active.sample }
+    let!(:messages_inbound1) { create_list :text_message, inbound1_count, send_at: now - 1.day, inbound: true, reporting_relationship: user1.reporting_relationships.active.sample }
+    let!(:messages_outbound1) { create_list :text_message, outbound1_count, send_at: now - 2.days, inbound: false, reporting_relationship: user1.reporting_relationships.active.sample }
+    let!(:transfers2) { create_list :text_message, 2, send_at: now - 1.day, type: Message::MARKER_TRANSFER, reporting_relationship: user2.reporting_relationships.active.sample }
+    let!(:messages_inbound2) { create_list :text_message, inbound2_count, send_at: now - 3.days, inbound: true, reporting_relationship: user2.reporting_relationships.active.sample }
+    let!(:messages_outbound2) { create_list :text_message, outbound2_count, send_at: now - 4.days, inbound: false, reporting_relationship: user2.reporting_relationships.active.sample }
```
Comment (tmaybe) [inline spec/models/department_spec.rb]: We should be able to use the factories to create these other message types rather than setting type, right?

**Review by tmaybe — CHANGES_REQUESTED** [review]

File: spec/models/department_spec.rb (hunk 4 above)
Comment (glassresistor) [reply spec/models/department_spec.rb]: Yes i just missed these

File: app/views/court_reminders/_court_reminders.erb (hunk 2 above)
Comment (glassresistor) [reply app/views/court_reminders/_court_reminders.erb]: surprised this didnt error

File: spec/factories/text_message.rb (hunk 3 above)
Comment (glassresistor) [reply spec/factories/text_message.rb:12]: yeah i just wanted to change as little as possible

**Pull request merged**